src/ORM/DataList.php

```diff
@@ -1115,6 +1116,11 @@ private function fetchEagerLoadHasOne(
         string $relationName,
         string $relationType
     ): array {
+        // Throw exception if developers try to filter/sort a has_one relation
+        if ($this->eagerLoadAllRelations[$relationChain] !== null) {
+            throw new LogicException("Cannot filter/sort $relationType relation $relationName");
+        }
```

Comment:
Say something like "transform" or "manipulate" instead of "filter/sort", since developers could be trying to do just about anything in the callback.